cmake/InitGitSubmodule.cmake

```diff
@@ -63,20 +63,26 @@ function(git_submodule_init)
 
   # actual code
   if(NOT EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/${GIT_SM_CHECK_FILE}")
-    message(
-      STATUS
-        "git submodule for ${GIT_SM_READABLE_NAME} missing from source code, will attempt to get it..."
-    )
-    execute_process(
-      COMMAND ${GIT_EXECUTABLE} submodule update --init
-              "${GIT_SM_SUBMODULE_PATH}"
-      TIMEOUT 30
+    set(GIT_SM_RECURSIVE_COMMAND "")
+    set(GIT_SM_TIMEOUT 30)
+    if(GIT_SM_RECURSIVE)
+      set(GIT_SM_RECURSIVE_COMMAND "--recursive")
+      # Use a longer timeout when also dealing with submodules of this submodule
+      set(GIT_SM_TIMEOUT 60)
+      message(STATUS "git submodule for ${GIT_SM_READABLE_NAME} missing from source code, will attempt to get it and, recursively, any submodules...")
```

Comment:
Can we output the first part of the message before the if and output another string if it's recursive? That way we don't need to duplicate the first part.

Reply:
:man_shrugging: Well I was looking to output a single message in either case and thought it was clearer to have reasonably similar ones for both cases.